smslibrary/src/main/java/com/eis/smslibrary/SMSHandler.java

```diff
@@ -74,12 +74,26 @@ public static SMSHandler getInstance() {
     /**
      * Setup for the handler.
      *
-     * @param context current context.
+     * @param context current context. Must not be null
+     * @throws NullPointerException If the given context is null
      */
     public void setup(Context context) {
```

Comment:
Add a @NonNullable too, so that we're 100% safe and sound

Reply:
Adding NonNull automatically throws the error? Didn't know, gonna use it more often. I'll also remove the if then. I'll leave the @throws just in case though.

Reply:
Don't think it throws the error automatically, the IDE warns the user that, if not correctly checked, the parameter might be null and there might be some problems in the method.

Reply:
That would be like having it in the specifications then, how should we go about this?

Reply:
(Can confirm, the error is not thrown)

Reply:
It's up to you, we now have narrower preconditions so if someone ever had to pass a null context, when an exception would appear they'd have to go deeper to find out that this method doesn't accept null.

Reply:
I'll leave both the code for the null check and the @nonnull notation.
What might happen is:
- a user purposely inserts null (in which case the IDE warns the user to not do it)
- a run-time error inserts a null value, breaking the system. The null check will catch that and stop execution instead of going forward as if nothing happened.